### PR TITLE
uniform and fixed micromenu

### PR DESCRIPTION
**Pull request by wuan**
This hopfully makes the micromenu more usable. For example some file viewing plugins do not contain the format part and the order of the items is not the same in all cases.

This PR sets the micromenu `View` to be used for all views showing file contents (e. g. View, Copyright, Audit) and the micromenu `View-Meta` to be used for views showing meta-information of a file.

Currently in the `View` case:
> One-Shot Copyright/Email/URL | One-Shot License   •   Info | View | Audit | Copyright/Email/Url/Author | ECC | Bucket   •   Hex | Text | Formatted   •   Refresh

And in the `View-Meta` case:
> One-Shot Copyright/Email/URL | One-Shot License   •   Info | View | Audit | Copyright/Email/Url/Author | ECC   •   Refresh

Basically, the view settings are missing here.

The menu generating code inside the plugins could still be abstracted. Discussion is welcome.

### CODE REVIEW

File: src/lib/php/common-menu.php [hunk 1]
```diff
@@ -196,11 +197,11 @@ function menu_insert_r(&$menuItems, $path, $pathRemainder, $LastOrder, $Target,
   if ($menuItemsExist) {
     foreach($menuItems as &$menuItem) {
       // need to escape the [ ] or the string will not match
-      if (!strcmp($menuItem->Name, $pathElement) && strcmp($menuItem->Name, "\\[BREAK\\]")) {
```
Comment (wuan) [inline src/lib/php/common-menu.php]: Did this really work as intended before? Before the change only one [BREAK] element per submenu was possible.

File: src/www/ui/ui-browse-license.php [hunk 2]
```diff
@@ -77,6 +77,7 @@ public function __construct() {
    */
   function RegisterMenus()
   {
+    return 0;
```
Comment (maximilianhuber) [inline src/www/ui/ui-browse-license.php]: Do you really want to `return 0;` here?

Comment (wuan) [inline src/www/ui/ui-browse-license.php]: No, thanks!

**Comment by maximilianhuber** [comment]
When I try your changes, I get the following errors:
tmp

**Comment by wuan** [comment]
Can you try a full install? The service.xml file needs to be updated.

**Comment by maximilianhuber** [comment]
I have used
```
$  sudo make uninstall && make && sudo make install
```
on root level to install your changes.

Where is the service.xml located?

**Comment by maximilianhuber** [comment]
After `vagrant up` with your fork, I get:
tmp
after adding the following lines to `src/www/ui/index.php`
```
ini_set('display_errors', 'On');
error_reporting(E_ALL | E_STRICT);
```

**Comment by wuan** [comment]
The script called by `fo-postinstall` seemed to dislike the dash in the dependency injection key. That is fixed now and the error message during `vagrant up` should be gone.

The depenency injection configuration file is located at `src/lib/php/services.xml`. It should be regenerated by make from the template file `src/lib/php/services.xml` through the following lines in the Makefile:
```
# generate services.xml
services.xml: services.xml-process
```
To be sure, that the file is updated, you can try
```
$  sudo make uninstall && make clean && make && sudo make install
```

**Comment by steffen-weber** [comment]
Other service ids are written in snake_case to solve the trouble with dashes.
The fatal error mentioned by @maximilianhuber was probably caused by caching which can be solved with ``sudo make empty-cache``.

**Comment by wuan** [comment]
Ah, I forgot the cache, what about cleaning the cache during `make install`?

**Comment by maximilianhuber** [comment]
I would support the idea with `empty-cache` while `make install`.

**Comment by maximilianhuber** [comment]
With the new modifications i am able to start FOSSology, but i have found an new problem with your current branch:
tmp
and the content of the table does not load via ajax:
tmp

### what i have done:
1. vagrant up with a virgin copy of your branch.
2. setup php (for the upload)
3. upload `openssl-1.0.2d.tar` and start some agents for it
4. click on browse -> license-browser

### modifications:
I still use 
```
ini_set('display_errors', 'On');
error_reporting(E_ALL | E_STRICT);
```

**Comment by wuan** [comment]
The missing variable is defined now. Please note that Ajax requests don't work because of the error display.
